lib/evaluateFunctions.js

```diff
@@ -96,4 +99,389 @@ function absoluteValue(node) {
+      }
+    }
+  }
+  else if (NodeType.isOperator(radicandNode) && radicandNode.op === '*') {
+    const children = [];
+    for (let i = 0; i < radicandNode.args.length; i++) {
+      const child = radicandNode.args[i];
+      if (NodeType.isParenthesis(child)) {
+        const grandChild = child.content;
+        if (grandChild.op === '*' && allEqual(grandChild.args)) {
+          const baseNode = child.content.args[0];
+          const exponentNode = NodeCreator.constant(rootValue);
+          children.push(NodeCreator.operator('^', [baseNode, exponentNode]));
+          continue;
+        }
+      }
+      children.push(child);
+    }
+
+    newNode.args[0] = NodeCreator.operator('*', children);
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.CONVERT_MULTIPLICATION_TO_EXPONENT, node, newNode);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given an nthRoot node with a multiplication under the radicand, will
+// distribute the nthRoot to all the arguments under the radicand as a dtep
+// e.g. nthRoot(2 * x^2, 2) -> nthRoot(2) * nthRoot(x^2)
+function distributeNthRoot(node) {
+  let newNode = clone(node, false);
+  const radicandNode = node.args[0];
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  const children = [];
+  for (let i = 0; i < radicandNode.args.length; i++) {
+    const child = radicandNode.args[i];
+    children.push(NodeCreator.nthRoot(child, rootNode));
+  }
+
+  newNode = NodeCreator.operator('*', children);
+  return NodeStatus.nodeChanged(
+    MathChangeTypes.DISTRIBUTE_NTH_ROOT, node, newNode);
+}
+
+// Given a multiplication node of nthRoots (with the same root)
+// will evaluate the nthRoot of each Node as a substep
+// e.g. nthRoot(2) * nthRoot(x^2) -> nthRoot(2) * x
+function evaluateNthRootForChildren(node) {
+  let newNode = clone(node, false);
+
+  const substeps = [];
+  for (let i = 0; i < newNode.args.length; i++) {
+    const child = newNode.args[i];
+    const childNodeStatus = nthRoot(child);
+    if (childNodeStatus.hasChanged()) {
+      newNode.args[i] = childNodeStatus.newNode;
+      substeps.push(NodeStatus.childChanged(newNode, childNodeStatus, i));
+    }
+  }
+
+  if (substeps.length > 0) {
+    return NodeStatus.nodeChanged(
+      MathChangeTypes.EVALUATE_DISTRIBUTED_NTH_ROOT, node, newNode, true, substeps);
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+// Given a multiplication nodes, including nthRoots, will combine the nodes
+// with the same radicand as a step
+// e.g. 2 * nthRoot(2) * nthRoot(x) -> 2 * nthRoot(2 * x)
+function combineRoots(node) {
+  let newNode = clone(node, false);
+  const rootNode = node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);
+
+  const children = [];
+  const radicandArgs = [];
+  for (let i = 0; i < newNode.args.length; i++) {
+    const child = newNode.args[i];
+    if (NodeType.isFunction(child, 'nthRoot') && child.args[1].equals(rootNode)) {
```

Comment:
what it doesn't have args[1] in the other case of  `node.args.length === 2 ? node.args[1] : NodeCreator.constant(2);`